src/renderer/util/index.ts

```diff
@@ -41,26 +62,66 @@ export async function determineFilesFromNestedPaths(
 ): Promise<string[]> {
   const filePaths = await Promise.all(
     paths.flatMap(async (fullPath) => {
-      const stats = await fsPromises.stat(fullPath);
-      if (!stats.isDirectory()) {
-        return [fullPath];
-      }
-      const canRead = await canUserRead(fullPath);
-      if (!canRead) {
-        throw new Error(`User does not have permission to read ${fullPath}`);
-      }
-      const pathsUnderFolder = await fsPromises.readdir(fullPath, {
-        withFileTypes: true,
-      });
-      return pathsUnderFolder
-        .filter((f) => f.isFile())
-        .map((f) => resolve(fullPath, f.name));
+      return determineFilesFromNestedPath(fullPath);
     })
   );
 
   return uniq(flatten(filePaths));
 }
 
+/**
+ * Use a given filepath's extension to determine if it is a "multifile".
+ * @param filePath Path to the file
+ */
+export function determineIsMultifile(filePath: string): boolean {
+  ///////////////////////////////////////////////////////////////////
+  // TODO - Remove this once multifile support is feature-complete //
+  //         Flip the boolean to "true" for testing                //
+  ///////////////////////////////////////////////////////////////////
+  const USE_MULTIFILE_ASSUMPTION = process.env.USE_MULTIFILE_ASSUMPTION !== undefined
+      ? process.env.USE_MULTIFILE_ASSUMPTION === 'true'
+      : false;
+  ///////////////////////////////////////////////////////////////////
+
+  const multifileExtensions = ['.zarr', '.sldy'];
+  const combinedExtensions = multifileExtensions.join('|');
+
+  // "ends with one of the listed extensions, ignoring casing"
+  // otherwise written like: /.zarr|.sldy$/i
+  const matcher = new RegExp(combinedExtensions + "$", 'i');
```

Comment:
Maybe the matching should be a different unit, so that it can be tested separately from the env var flag.

Reply:
I think I've got a few arguments against that:

* The env var flag is going away someday anyways
* There's not much going on in this function beyond the regex matching anyways, so any existing unit tests should cover the regex matching.
* This would be the only place where that method would get used - writing a separate test for it would basically be testing a private method, wouldn't it?

Reply:
> The env var flag is going away someday anyways.

But the regex will remain, right?  Im just talking about testing the regex part.

> There's not much going on in this function beyond the regex matching, so any existing unit tests should cover it.
> This would be the only place where that method would get used - writing a separate test for it would basically be testing a private method, wouldn't it?

Reply:
(we talked this throguh and decided the existing unit tests are sufficient)